### ISSUE TITLE
App won't start up on 3.32.0+

### ISSUE THREAD

**Issue by ammmze** [tolgee/tolgee-platform#1947]
I'm having an issue starting up the latest versions of the app. It almost starts up then crashes with the following error:

```

org.springframework.context.ApplicationContextException: Failed to start bean 'redisPubsubContainer'; nested exception is java.lang.NoClassDefFoundError: org/springframework/data/redis/connection/zset/Tuple
	at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:181) ~[spring-context-5.3.28.jar:5.3.28]
	at org.springframework.context.support.DefaultLifecycleProcessor.access$200(DefaultLifecycleProcessor.java:54) ~[spring-context-5.3.28.jar:5.3.28]
	at org.springframework.context.support.DefaultLifecycleProcessor$LifecycleGroup.start(DefaultLifecycleProcessor.java:356) ~[spring-context-5.3.28.jar:5.3.28]
	at java.base/java.lang.Iterable.forEach(Iterable.java:75) ~[na:na]
	at org.springframework.context.support.DefaultLifecycleProcessor.startBeans(DefaultLifecycleProcessor.java:155) ~[spring-context-5.3.28.jar:5.3.28]
	at org.springframework.context.support.DefaultLifecycleProcessor.onRefresh(DefaultLifecycleProcessor.java:123) ~[spring-context-5.3.28.jar:5.3.28]
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:937) ~[spring-context-5.3.28.jar:5.3.28]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:586) ~[spring-context-5.3.28.jar:5.3.28]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:147) ~[spring-boot-2.7.13.jar:2.7.13]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:731) ~[spring-boot-2.7.13.jar:2.7.13]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:408) ~[spring-boot-2.7.13.jar:2.7.13]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:307) ~[spring-boot-2.7.13.jar:2.7.13]
	at io.tolgee.Application$Companion.main(Application.kt:26) ~[app/:na]
	at io.tolgee.Application.main(Application.kt) ~[app/:na]
Caused by: java.lang.NoClassDefFoundError: org/springframework/data/redis/connection/zset/Tuple
	at org.redisson.spring.data.connection.RedissonConnection.<clinit>(RedissonConnection.java:2361) ~[redisson-spring-data-31-3.23.2.jar:3.23.2]
	at org.redisson.spring.data.connection.RedissonConnectionFactory.getConnection(RedissonConnectionFactory.java:113) ~[redisson-spring-data-31-3.23.2.jar:3.23.2]
	at org.springframework.data.redis.listener.RedisMessageListenerContainer$Subscriber.initialize(RedisMessageListenerContainer.java:1186) ~[spring-data-redis-2.7.13.jar:2.7.13]
	at org.springframework.data.redis.listener.RedisMessageListenerContainer.doSubscribe(RedisMessageListenerContainer.java:318) ~[spring-data-redis-2.7.13.jar:2.7.13]
	at org.springframework.data.redis.listener.RedisMessageListenerContainer.lazyListen(RedisMessageListenerContainer.java:295) ~[spring-data-redis-2.7.13.jar:2.7.13]
	at org.springframework.data.redis.listener.RedisMessageListenerContainer.lazyListen(RedisMessageListenerContainer.java:265) ~[spring-data-redis-2.7.13.jar:2.7.13]
	at org.springframework.data.redis.listener.RedisMessageListenerContainer.start(RedisMessageListenerContainer.java:249) ~[spring-data-redis-2.7.13.jar:2.7.13]
	at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:178) ~[spring-context-5.3.28.jar:5.3.28]
	... 13 common frames omitted
Caused by: java.lang.ClassNotFoundException: org.springframework.data.redis.connection.zset.Tuple
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:602) ~[na:na]
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178) ~[na:na]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[na:na]
	... 21 common frames omitted
```

I pulled down the source code and I see that we now have `redisson-spring-data-27` AND `redisson-spring-data-31` (previously we only had `-27`) in the project. I took a look at the `RedissonConnection` for both of those and only `-31` uses that `Tuple` class, so it does appear to be using `-31`, not `-27` (also confirmed by the stack trace). I noticed we have an [exclusion](https://github.com/tolgee/tolgee-platform/blob/373a602017aabff8d31cf5d1ca093473f1c6ea6c/backend/api/build.gradle#L62-L64) for `-31` in all the gradle modules EXCEPT the new `security` one that was added in 3.32.0. I suspect we need the same exclusion in the new `security` module?

**Comment by ammmze** [comment]
I should also add that in my dev kubernetes environment (still using the same official Tolgee docker image though) it somehow did come up. But in another environment (still kubernetes and very similar environment though perhaps a little more horsepower) I have not been able to get it to come up. Not sure if there is a race condition selecting which class to use from the class path, or perhaps filesystem order plays a role in the order classes are loaded, idk...but IntelliJ also shows that the `Tuple` class does not exist in the project, so I think `-31` just needs to be excluded like the other modules.

**Comment by JanCizmar** [comment]
Hey! You're right. We should add the exclusion too.

**Comment by JanCizmar** [comment]
It's fixed in v3.32.2

**Comment by ammmze** [comment]
Awesome! Thank you @JanCizmar!
